build-system/server.js

```diff
@@ -207,32 +200,10 @@ app.use('/min/', function(req, res) {
   proxyToAmpProxy(req, res, /* minify */ true);
 });
 
-function setAMPAccessControlHeader(res, path) {
-  var curUrl = url.parse(path, true);
-  if (curUrl.pathname.indexOf('/examples.build/analytics.config.json') > 0) {
-    res.setHeader('AMP-Access-Control-Allow-Source-Origin',
-        'http://localhost:' + port);
-  }
-}
-
-paths.split(',').forEach(function(pth) {
-  // Serve static files that exist
-  app.use(serveStatic(path.join(process.cwd(), pth),
-        {setHeaders: setAMPAccessControlHeader}));
-  // Serve directory listings
-  app.use(serveIndex(path.join(process.cwd(), pth),
-    {'icons':true,'view':'details'}));
-});
-
-// 404 everything else
-app.use(function notFound(req, res) {
-  var done = finalhandler(req,res);
-  var err = new Error('File Not Found');
-  err.status = 404;
-  done(err);
-});
-
-app.listen(port, function() {
-  console./*OK*/log('serving %s at http://localhost:%s', paths, port);
+app.use('/examples.build/analytics.config.json', function (req, res, next) {
+  res.setHeader('AMP-Access-Control-Allow-Source-Origin',
+      'http://' + req.headers.host);
```

Comment:
Yes. 

The ideal fix requires an upgrade from "connect" to "express" so that we can get the protocol info right from the request. I had a separate PR to address that. 

Meanwhile, there're a few other places don't work well with HTTPS yet. 

- We have many test examples hardcoded "http://localhost". I had a PR to fix that as well.
- Once a test page loaded via HTTPS, any HTTP resource load will throw exception. Will take a look at that later.